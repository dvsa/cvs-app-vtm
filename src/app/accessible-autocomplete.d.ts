declare module 'accessible-autocomplete/dist/accessible-autocomplete.min' {
  interface ElementParam {
    element: HTMLElement;
  }
  interface EnhancedElementParam {
    selectElement: HTMLElement | null;
  }
  interface DefaultParams {
    id?: string;
    source?: Array<any>;
    onConfirm?: (value: string) => void;
    confirmOnBlur?: boolean;
    required?: true;
    defaultValue?: string;
    autoselect?: boolean;
    showAllValues?: boolean;
    dropdownArrow?: Function;
  }

  export interface AutocompleteParams extends DefaultParams, ElementParam {}
  export interface AutocompleteEnhanceParams extends DefaultParams, EnhancedElementParam {}

  export default function accesibleAutocomplete(params: AutocompleteParams): void;
  export function enhanceSelectElement(params: AutocompleteEnhanceParams): void;
}
