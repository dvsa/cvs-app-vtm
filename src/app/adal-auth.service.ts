import { Injectable } from '@angular/core';
import { MsAdalAngular6Service } from 'microsoft-adal-angular6';
// import { MsAdalAngular6Module } from 'microsoft-adal-angular6';


@Injectable({
  providedIn: 'root'
})
export class AdalAuthService {

  constructor(
    private msAdal: MsAdalAngular6Service
  ) { }

  isAuthenticated() {
    return this.msAdal.isAuthenticated
  }

  getProfile() {
    return this.msAdal.userInfo.profile
  }

  getResourceForEndpoint(url) {
    return this.msAdal.GetResourceForEndpoint(url)
  }

  acquireToken(resource) {
    return this.msAdal.acquireToken(resource)
  }

  accessToken() {
    return this.msAdal.accessToken
  }
}
