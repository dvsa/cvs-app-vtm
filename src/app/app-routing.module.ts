import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { MsalGuard } from '@azure/msal-angular';
import { TitleResolver } from './resolvers/title/title.resolver';

const routes: Routes = [
  {
    path: '',
    resolve: { title: TitleResolver },
    children: [
      {
        path: '',
        data: { title: 'Home' },
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/home/home.module').then(m => m.HomeModule)
      },
      {
        path: 'search',
        data: { title: 'Technical record search' },
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/search/search.module').then(m => m.SearchModule)
      },
      {
        path: 'test-records',
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/test-records/test-records.module').then(m => m.TestRecordsModule)
      },
      {
        path: 'tech-records',
        canActivate: [MsalGuard],
        loadChildren: () => import('./features/tech-record/tech-record.module').then(m => m.TechRecordsModule)
      }
    ]
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes, { onSameUrlNavigation: 'reload', scrollPositionRestoration: 'enabled' })],
  exports: [RouterModule]
})
export class AppRoutingModule {}
