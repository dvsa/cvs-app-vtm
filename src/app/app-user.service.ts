import { Injectable } from '@angular/core';

import { UserDetails } from './models/user-details';

@Injectable({ providedIn: 'root' })
export class UserService {
  private user: UserDetails;

  constructor() {}

  setUser(userProfile) {
    this.user = {} as UserDetails;
    this.user.msOid = userProfile.oid;
    this.user.msUser = userProfile.unique_name;
  }

  // TODO: Refactor, remove when auth class is implemented,
  // user retrieved from the store user.details
  getUser() {
    return this.user;
  }

}
