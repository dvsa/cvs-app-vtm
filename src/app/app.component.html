<app-header [username]="userName" (logOutEvent)="logOut()"></app-header>

<div class="govuk-width-container">
  <main class="govuk-main-wrapper" id="main-content" role="main">
    <h1 class="govuk-heading-xl">Default page template</h1>
  </main>
</div>

<app-footer></app-footer>
