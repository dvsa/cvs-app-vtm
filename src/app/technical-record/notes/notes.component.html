<dl class="govuk-summary-list">
  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'hgv' || activeRecord.vehicleType === 'trl'">
    <dt class="govuk-summary-list__key">Notes</dt>
    <dd id="test-notes" class="govuk-summary-list__value">
      <span id="test-notes" class="notes-section">{{ activeRecord.notes | DefaultNullOrEmpty }}</span>
    </dd>
  </div>

  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
    <dt class="govuk-summary-list__key">Remarks</dt>
    <dd id="test-dispensations" class="govuk-summary-list__value">
      <span id="test-notes" class="notes-section">{{ activeRecord.remarks | DefaultNullOrEmpty }}</span>
    </dd>
  </div>

  <div class="govuk-summary-list__row" *ngIf="activeRecord.vehicleType === 'psv'">
    <dt class="govuk-summary-list__key">Dispensations</dt>
    <dd id="test-dispensations" class="govuk-summary-list__value">
      <span id="test-notes" class="notes-section">{{ activeRecord.dispensations | DefaultNullOrEmpty }}</span>
    </dd>
  </div>
</dl>
