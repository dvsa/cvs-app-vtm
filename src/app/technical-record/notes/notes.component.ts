import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';
import { TechRecord } from '@app/models/tech-record.model';

@Component({
  selector: 'vtm-notes',
  templateUrl: './notes.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class NotesComponent implements OnInit {
  @Input() editState: boolean;
  @Input() activeRecord: TechRecord;

  notes: string;

  constructor() {}

  ngOnInit() {
    this.notes = this.activeRecord.notes;
  }
}
