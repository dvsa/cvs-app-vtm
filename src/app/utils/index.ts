export * from './mockStore';
export * from './test-model.utils';
export * from './testing.utils';
export * from './form.utils';


