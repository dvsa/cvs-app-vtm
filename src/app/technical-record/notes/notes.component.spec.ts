import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';

import { TESTING_UTILS } from '@app/utils/testing.utils';
import { NotesComponent } from './notes.component';
import { SharedModule } from '@app/shared';
import { VEHICLE_TYPES } from '@app/app.enums';

describe('NotesComponent', () => {
  let component: NotesComponent;
  let fixture: ComponentFixture<NotesComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [NotesComponent],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(NotesComponent);
    component = fixture.componentInstance;
  });

  it('should create view only for hgv and trl with populated data', () => {
    component.activeRecord = TESTING_UTILS.mockTechRecord({
      notes: 'some notes',
      vehicleType: VEHICLE_TYPES.HGV
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should create view only for psv populated data', () => {
    component.activeRecord = TESTING_UTILS.mockTechRecord({
      remarks: 'some remarks',
      dispensations: '10/10',
      vehicleType: VEHICLE_TYPES.PSV
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should create view only for motorcycle populated date', () => {
    component.activeRecord = TESTING_UTILS.mockTechRecord({
      notes: 'some remarks',
      dispensations: '10/10',
      vehicleType: VEHICLE_TYPES.Moto
    });

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
