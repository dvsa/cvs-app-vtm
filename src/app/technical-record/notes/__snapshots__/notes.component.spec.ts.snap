// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesComponent should create view only for hgv and trl with populated data 1`] = `
<vtm-notes
  activeRecord={[Function Object]}
>
  <dl
    class="govuk-summary-list"
  >
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Notes
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-notes"
      >
        <span
          class="notes-section"
        >
          some notes
        </span>
      </dd>
    </div>
    
    
  </dl>
</vtm-notes>
`;

exports[`NotesComponent should create view only for psv populated data 1`] = `
<vtm-notes
  activeRecord={[Function Object]}
>
  <dl
    class="govuk-summary-list"
  >
    
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Remarks
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-remarks"
      >
        <span
          class="notes-section"
        >
          some remarks
        </span>
      </dd>
    </div>
    
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Dispensations
      </dt>
      <dd
        class="govuk-summary-list__value"
        id="test-dispensations"
      >
        <span
          class="notes-section"
        >
          -
        </span>
      </dd>
    </div>
  </dl>
</vtm-notes>
`;
